src/config/config.c

```diff
@@ -398,45 +398,51 @@ static void initConfig(struct config *conf)
 
 	// struct dns
 	conf->dns.upstreams.k = "dns.upstreams";
-	conf->dns.upstreams.h = "Array of upstream DNS servers used by Pi-hole\n Example: [ \"8.8.8.8\", \"127.0.0.1#5335\", \"docker-resolver\" ]";
-	conf->dns.upstreams.a = cJSON_CreateStringReference("array of IP addresses and/or hostnames, optionally with a port (#...)");
+	conf->dns.upstreams.h = "Upstream DNS Servers to be used by Pi-hole. If this is not set, Pi-hole will not resolve any non-local queries.\n\n Example: [ \"8.8.8.8\", \"127.0.0.1#5335\", \"docker-resolver\" ]";
+	conf->dns.upstreams.a = cJSON_CreateStringReference("Array of IP addresses and/or hostnames, optionally with a port (#...)");
 	conf->dns.upstreams.t = CONF_JSON_STRING_ARRAY;
 	conf->dns.upstreams.d.json = cJSON_CreateArray();
 	conf->dns.upstreams.f = FLAG_RESTART_FTL;
 	conf->dns.upstreams.c = validate_stub; // Type-based checking + dnsmasq syntax checking
 
 	conf->dns.CNAMEdeepInspect.k = "dns.CNAMEdeepInspect";
 	conf->dns.CNAMEdeepInspect.h = "Use this option to control deep CNAME inspection. Disabling it might be beneficial for very low-end devices";
+	conf->dns.CNAMEdeepInspect.a = cJSON_CreateStringReference("<true|false>");
```

Comment:
Is this the best way to show this? I'm asking because, sometimes, we just show a human-(not even machine!)-readable string here what to do. So we may better put here `true or false`? Just a suggestion.

Otherwise, it'd be cleaner to avoid these 80 *individual* additions of `conf->x.y.a = cJSON_CreateStringReference("<true|false>");` and instead put once into the verification loop at the end of this function:

``` c
if(conf_item.t == CONF_BOOL && conf_item.a == NULL)
	conf_item.a = cJSON_CreateStringReference("true or false");
```

This removes code duplication and ensures we are not overwriting any possibly manually set additional hints.

Reply:
OK, I agree with that. I had `true/false` before but thought it looked weird. And then I went down a whole rabbit hole of backticking.

I'll take your suggestion here though, it makes it more maintainable